ledger/src/blockstore_processor.rs

```diff
@@ -111,268 +78,198 @@ fn first_err(results: &[Result<()>]) -> Result<()> {
     Ok(())
 }
 
-// Includes transaction signature for unit-testing
-fn get_first_error(
-    batch: &TransactionBatch,
-    fee_collection_results: Vec<Result<()>>,
-) -> Option<(Result<()>, Signature)> {
-    let mut first_err = None;
-    for (result, transaction) in fee_collection_results
+// for each index, builds a transaction dependency graph of indices that need to execute before
+// the current one.
+// The returned Vec<HashSet<usize>> is a 1:1 mapping for the indices that need to be executed
+// before that index can be executed
+fn build_dependency_graph(tx_account_locks: &[TransactionAccountLocks]) -> Vec<IntSet<usize>> {
```

Comment:
@buffalu just hi. :)

fyi, i just copied this dep graph logics to https://github.com/solana-labs/solana/pull/31239#discussion_r1205565836 for bench purposes (not on production code-path, for now)

that massive pr contains bigger-yet-optimized `BankTransactionExecutor` (i.e. thread pool for executing unbatched txes), imho. so, you might be interested.

that pr's general mechanism will be plumbed with banking stage. so, it's too early for you to comment on it, but i hope any jito-specific block ordering with bare Txes and Bundles will be more straight forward to implement (including git rebases)